### ISSUE TITLE
Updating a read-only TypedDict item via `.update(…)` should be forbidden

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#3098]
Emit a diagnostic if "a TypedDict with a read-only item is updated with another TypedDict that declares that key.". See [this conformance test](https://shark.fish/typing-conformance-report/typeddicts_readonly_update.html#L23):
```py
class A(TypedDict):
    x: ReadOnly[int]
    y: int


a1: A = {"x": 1, "y": 2}
a2: A = {"x": 3, "y": 4}
a1.update(a2)  # E

```
